app/models/spree/shipping_method.rb

```diff
@@ -13,11 +13,11 @@ class ShippingMethod < ApplicationRecord
 
     default_scope -> { where(deleted_at: nil) }
 
-    has_many :shipping_rates, inverse_of: :shipping_method
+    has_many :shipping_rates, inverse_of: :shipping_method, dependent: :destroy
     has_many :shipments, through: :shipping_rates
-    has_many :shipping_method_categories
-    has_many :shipping_categories, through: :shipping_method_categories
-    has_many :distributor_shipping_methods
+    has_many :shipping_method_categories, dependent: :destroy
+    has_many :shipping_categories, through: :shipping_method_categories, dependent: nil
```

Comment:
It seems to be that the `dependent` option is irrelevant for `has_many through:`  associations.
We've already specified above that the join table record should get destroyed, and the `shipping_categories` record doesn't have a foreign key in it. 

I tried removing this option and Rubocop doesn't complain. This is consistent with the below `has_many through` association.

```suggestion
    has_many :shipping_categories, through: :shipping_method_categories
```